odoo/http.py

```diff
@@ -1059,6 +1059,14 @@ def flatten(self):
             self.response.append(self.render())
             self.template = None
 
+    def set_cookie(self, *args, **kwargs):
```

Comment:
Why are both implementations of `set_cookie` in this file not written the same way?
On top of not being the same (but doing the same), one is fallbacking on None while the other is fallbacking on 'required' (as it should, as per the js def)

Also, why don't we just do nothing if not allowed, instead of removing it (max_age/expires = 0) ?
I mean, why not, but is it really needed? It's weird to have it behaves differently than the js.

Reply:
Unifying both python and the js implementation in https://github.com/odoo/odoo/pull/101859/commits/99b5b753b61383d356675687cb6cd67e88557d1f or https://github.com/odoo/odoo/pull/101859.